app/Http/Controllers/ProductsController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Service\ProductService;
+use Carbon\Carbon;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+use Illuminate\Support\Collection;
+
+class ProductsController extends Controller
+{
+    private ProductService $productService;
+
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
+
+    public function getProducts(): JsonResponse
+    {
+        $data = $this->productService->parseProducts();
+        /** @var Collection $products */
+        $products = $data['products'];
+
+        $firstHalfRevenue = $products->filter(function ($item) {
+            /** @var Carbon $date */
+            $date = $item[ProductService::DATE_ADDED_COLUMN];
+            return $date->day <= 15;
+        })->sum(ProductService::PRICE_COLUMN);
+
+        $secondHalfRevenue = $products->filter(function ($item) {
+            /** @var Carbon $date */
+            $date = $item[ProductService::DATE_ADDED_COLUMN];
+            return $date->day > 15;
+        })->sum(ProductService::PRICE_COLUMN);
```

Comment:
As a suggestion, there is Collection::partition() that does the same thing in one go:
```suggestion
        [$firstHalfRevenue, $secondHalfRevenue] = $products->partition(function ($item) {
            /** @var Carbon $date */
            $date = $item[ProductService::DATE_ADDED_COLUMN];
            return $date->day <= 15;
        })->map(fn($halfRevenue) => $halfRevenue->sum(ProductService::PRICE_COLUMN));
```